namespaces/live.cloud-platform.service.justice.gov.uk/hmpps-restricted-patients-api-dev/resources/hmpps-restricted-patients-domain-events-queue.tf

```diff
@@ -104,10 +104,11 @@ resource "aws_sns_topic_subscription" "restricted_patients_queue_for_domain_even
   topic_arn = data.aws_ssm_parameter.hmpps-domain-events-topic-arn.value
   protocol  = "sqs"
   endpoint  = module.restricted_patients_queue_for_domain_events.sqs_arn
+  filter_policy_scope = "MessageBody"
   filter_policy = jsonencode({
-    eventType = [
-      "prison-offender-events.prisoner.merged",
-      "prisoner-offender-search.prisoner.released"
+    "$or": [
+      { "additionalInformation.eventType": [ "prison-offender-events.prisoner.merged" ] },
+      { "additionalInformation.eventType": [ "prisoner-offender-search.prisoner.released" ], "additionalInformation.reason": [ "RELEASED" ] }
```

Comment:
The logic I'm after is (after several syntax errors!):
`eventType == merged || (eventType == released and reason == RELEASED)`

Hope that `additionaLInformation.eventType` works.